src/main/java/org/apache/fineract/api/BatchApi.java

```diff
@@ -143,6 +151,15 @@ private BatchDTO generateDetails (Batch batch) {
         BigDecimal failedAmount = BigDecimal.ZERO;
 
         for (Transfer transfer : transfers) {
+            Variable variable = variableRepository.findByWorkflowInstanceKeyAndVariableName("paymentMode",
```

Comment:
use `Optional<Variable>` and `isPresent`

Reply:
done